data/src/main/java/com/microsoft/azure/kusto/data/ConnectionStringBuilder.java

```diff
@@ -94,6 +96,17 @@ public static ConnectionStringBuilder createWithAadApplicationCredentials(String
         return createWithAadApplicationCredentials(resourceUri, applicationClientId, applicationKey, null);
     }
 
+    public static ConnectionStringBuilder createWithAadApplicationCredentials(String resourceUri,
+                                                                              String applicationClientId,
+                                                                              String applicationKey,
+                                                                              String authorityId,
+                                                                              String originApplicationVersion)
+    {
```

Comment:
originApplicationVersion is not related to authentication.
Please add a setter and use it in order to change it.

Reply:
Please remove originApplicationVersion from the constructor.